src/solver/optimisation/adequacy_patch_csr/construct_problem_constraints_LHS.cpp

```diff
@@ -0,0 +1,305 @@
+                           int* Colonne)
+{
+    int hour = hourlyCsrProblem.hourInWeekTriggeredCsr;
+    int Var;
+    int NombreDeTermes;
+    int Interco;
+    int NbInterco;
+    double Poids;
+    PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre;
+    const CONTRAINTES_COUPLANTES* MatriceDesContraintesCouplantes;
+    std::string NomDeLaContrainte;
+
+    ProblemeAResoudre = ProblemeHebdo->ProblemeAResoudre;
+
+    // Special case of the binding constraints
+    for (int CntCouplante = 0; CntCouplante < ProblemeHebdo->NombreDeContraintesCouplantes;
+         CntCouplante++)
+    {
+        MatriceDesContraintesCouplantes
+          = ProblemeHebdo->MatriceDesContraintesCouplantes[CntCouplante];
+
+        if (MatriceDesContraintesCouplantes->TypeDeContrainteCouplante == CONTRAINTE_HORAIRE)
+        {
+            NbInterco
+              = MatriceDesContraintesCouplantes->NombreDInterconnexionsDansLaContrainteCouplante;
+            NombreDeTermes = 0;
+            for (int Index = 0; Index < NbInterco; Index++)
+            {
+                Interco = MatriceDesContraintesCouplantes->NumeroDeLInterconnexion[Index];
+                Poids = MatriceDesContraintesCouplantes->PoidsDeLInterconnexion[Index];
+
+                if (ProblemeHebdo->adequacyPatchRuntimeData.originAreaMode[Interco]
+                      == Data::AdequacyPatch::physicalAreaInsideAdqPatch
+                    && ProblemeHebdo->adequacyPatchRuntimeData.extremityAreaMode[Interco]
+                         == Data::AdequacyPatch::physicalAreaInsideAdqPatch)
+                {
+                    Var = ProblemeHebdo->CorrespondanceVarNativesVarOptim[hour]
+                            ->NumeroDeVariableDeLInterconnexion[Interco];
+
+                    if (Var >= 0)
+                    {
+                        Pi[NombreDeTermes] = Poids;
+                        Colonne[NombreDeTermes] = Var;
+                        NombreDeTermes++;
+                    }
+                }
+            }
+
+            if (NombreDeTermes > 0) // current binding constraint contains an interco type 2<->2
+            {
+                hourlyCsrProblem.numberOfConstraintCsrHourlyBinding[CntCouplante]
+                  = ProblemeAResoudre->NombreDeContraintes;
+
+                NomDeLaContrainte = "bc::hourly::" + std::to_string(hour) + "::"
+                                    + MatriceDesContraintesCouplantes->NomDeLaContrainteCouplante;
+
+                logs.debug() << "C (bc): " << ProblemeAResoudre->NombreDeContraintes << ": "
+                             << NomDeLaContrainte;
+
+                OPT_ChargerLaContrainteDansLaMatriceDesContraintes(
+                  ProblemeAResoudre,
+                  Pi,
+                  Colonne,
+                  NombreDeTermes,
+                  MatriceDesContraintesCouplantes->SensDeLaContrainteCouplante);
+            }
+        }
+    }
+}
+
+void OPT_ConstruireLaMatriceDesContraintesDuProblemeQuadratique_CSR(
```

Comment:
According to review of : 
- **adequacy_patch.cpp**, 
- **sim_structure_probleme_economique.h**

this function should be a method of a class  **csrQuadraticProblem**, and have no argument.